### DISCUSSION TITLE
Running per spec with tag exclusion

### DISCUSSION THREAD

**Discussion by jonathangiber** [getgauge/gauge#2135] [General]
Hi,
I would like to know if when running per spec (gauge run example.spec), is there any way to exclude per tag?

For example following command wouldn't be able to run this spec:

gauge run example.spec --tags !in-progress

Example specification
----------------------
Tags: in-progress

* Example step

Can explain the motivation behind it if needed.
Thanks

**Comment by zbrea** [comment]
Try:
gauge run specs/example.spec --tags "!in-progress"
or
gauge run --tags "!in-progress" specs/example.spec
or 
gauge run --tags "!in-progress" specs

**Comment by zbrea** [comment]
I found this info in the gauge doc :
![image](https://user-images.githubusercontent.com/59898499/138602523-e73fbbe1-cb66-4a68-98ea-89c8f9799d8e.png)

**Comment by jonathangiber** [comment]
Hi,
Unfortunately this solution doesn't work, when executing per spec, it would seem that gauge completely ignores the tags, any chance to confirm? @sriv @zabil

**Reply by sriv** [reply]
Need to check this, will check once I get access to my computer

**Reply by zabil** [reply]
Can you please replicate this in the sample code? For example I tried this and it works on the template

```
➜  gauge init js_simple
➜  ls
env  logs  manifest.json  package-lock.json  package.json  reports  sahaj.js  specs  tests  todo.js
➜  gauge list --tags
[Tags]
single word

➜  vowels gauge run --tags='!Single word' specs/example.spec
# Specification Heading
  ## Vowel counts in multiple word	 ✔ ✔

Successfully generated html-report to =>./reports/html-report/index.html

Specifications:	1 executed	1 passed	0 failed	0 skipped
Scenarios:	1 executed	1 passed	0 failed	0 skipped

Total time taken: 90ms
➜  gauge run specs/example.spec
# Specification Heading
  ## Vowel counts in single word	 ✔ ✔
  ## Vowel counts in multiple word	 ✔ ✔

Successfully generated html-report to => ./html-report/index.html

Specifications:	1 executed	1 passed	0 failed	0 skipped
Scenarios:	2 executed	2 passed	0 failed	0 skipped

Total time taken: 63ms
➜  vowels
```

**Comment by jonathangiber** [comment]
It was the order of the parameters indeed, working as expected, thanks
